cclib/parser/gamessdatparser.py

```diff
@@ -1,246 +1,314 @@
+                    mass        = float(parts[1])
+                    coordinates = [float(coord) for coord in parts[2:]]
+
+                    assert len(coordinates) == 3
+
+                    self.atommasses.append(mass)
+                    self.atomcoords.append(coordinates)
+                elif len(parts) == 2:
+                    basis_set = parts[0] + '-' + parts[1] + 'G'
+                    self.metadata['basis_set'] = basis_set
+
+                    
+                line = next(inputfile)
+
+
+        # Extract energy
+
+        # --- CLOSED SHELL ORBITALS --- GENERATED AT Mon Aug  5 13:05:47 2019
+        # water                                                                           
+        # E(RHF)=      -74.9643287920, E(NUC)=    8.8870072224,   13 ITERS
+
+        # Extract E(RHF) value
+
+        if "E(R" in line:
+            val_pattern = r"E\(R[^,]+"
+            match = re.search(val_pattern, line)
+            val = float(match.group().split(' ')[-1])
+            self.scfenergies = [ val ]
+
+        # Extract E(NUC) value 
+
+        if "E(NUC)=" in line:
+            pattern_e_nuc = r"E\(NUC[^,]+"
+            match_e_nuc = re.search(pattern_e_nuc, line)
+            nuc_value = float(match_e_nuc.group().split(' ')[-1].strip())
+            self.metadata["E_NUC"] = nuc_value
+
+        # Extract number of ITERS 
+
+        # if "ITERS" in line:
+        #     iters_value = int(line.split()[-1])
+        #     self.metadata["ITERS"] = iters_value
+
+        # Extract vectors
+
+        #  $VEC   
+        #  1  1 9.94202990E-01 2.59157151E-02 2.40311554E-03 3.18904296E-03 0.00000000E+00
+        #  1  2-5.62726478E-03-5.62726567E-03
+        #  2  1-2.34217935E-01 8.45881798E-01 7.04411127E-02 9.34785483E-02-0.00000000E+00
+        #  2  2 1.56449309E-01 1.56449336E-01
+        #  3  1-1.17687509E-08 6.36837896E-08 4.81820843E-01-3.63078078E-01 0.00000000E+00
+        #  3  2 4.46376801E-01-4.46376807E-01
+        #  4  1 1.00458159E-01-5.21395067E-01 4.65965488E-01 6.18357071E-01 0.00000000E+00
+        #  4  2 2.89063958E-01 2.89063907E-01
+        #  5  1-0.00000000E+00-0.00000000E+00-0.00000000E+00-0.00000000E+00 1.00000000E+00
+        #  5  2-0.00000000E+00-0.00000000E+00
+        #  6  1-1.28350522E-01 8.32525679E-01 4.40905546E-01 5.85101037E-01-0.00000000E+00
+        #  6  2-7.75800880E-01-7.75800504E-01
+        #  7  1 3.90260255E-08-2.79856684E-07 7.79855187E-01-5.87663268E-01 0.00000000E+00
+        #  7  2-8.08915389E-01 8.08915850E-01
+        #  $END   
+
+        # Extract vector information and populate mocoeffs.
+        # Also extract nbasis from here.
+
+        if line[1:5] == "$VEC":
+
+            self.nbasis   = None
+            self.nmo      = 0
+            self.mocoeffs = []
+
+            line = next(inputfile)
+
+            while "$END" not in line:
+                
+                moc_line = line.replace('-', ' -').replace('E -', 'E-').strip()

```

Comment:
It took me a moment to understand what this does.  If it works, it's fine.  What I had suggested was using
```python
line[0:2], line[2:5], line[5:5+15], line[20:20+15], line[35:35+15], line[50:50+15], line[65:65+15]
```
since the width of each field is fixed.